Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/ChromeDriverEx.cs

```diff
@@ -26,36 +26,27 @@ public static class ChromeDriverEx
     {
         public static Screenshot GetFullPageScreenshot(this ChromiumDriver driver)
         {
-            //Dictionary will contain the parameters needed to get the full page screen shot
+            // Capture the original scroll position
+            Dictionary<string, object> originalScrollPosition = (Dictionary<string, object>)driver.ExecuteScript("return { x: window.pageXOffset, y: window.pageYOffset };");
+
+            // Capture page dimensions and device metrics
             Dictionary<string, Object> metrics = new Dictionary<string, Object>
             {
-                ["width"] = driver.ExecuteScript("return Math.max(window.innerWidth,document.body.scrollWidth,document.documentElement.scrollWidth)"),
-                ["height"] = driver.ExecuteScript("return Math.max(window.innerHeight,document.body.scrollHeight,document.documentElement.scrollHeight)")
+                ["width"] = driver.ExecuteScript("return Math.max(window.innerWidth, document.body.scrollWidth, document.documentElement.scrollWidth)"),
+                ["height"] = driver.ExecuteScript("return Math.max(window.innerHeight, document.body.scrollHeight, document.documentElement.scrollHeight)"),
+                ["deviceScaleFactor"] = Convert.ToDouble(driver.ExecuteScript("return window.devicePixelRatio") ?? 1),
+                ["mobile"] = driver.ExecuteScript("return typeof window.orientation !== 'undefined'")
             };
-            object devicePixelRatio = driver.ExecuteScript("return window.devicePixelRatio");
-            if (devicePixelRatio != null)
```

Comment:
Let's keep the original code, it was added after JS returned inconsistent values in long or sometimes in double. and long fails to case to double.

Reply:
Made changes - see latest commit.